### PR TITLE
Added --dev feature for creating a origin feature branch into upstream feature branch pull request

### PR DESCRIPTION
**Pull request by RobertReaves**
### Short description of the work completed

> Added a new --dev flag to allow origin to upstream feature branch pull requests. In fact I'm using it now!

### Steps to test (if not obvious)

1. Create a local feature branch.
2. Do some work and commit it.
3. Your work is done!
4. Run `tag-release --dev` and follow the instructions! (This might break your origin and upstream are the same...)

### For Reviewer Use Only

- [x] Code Reviewed
- [x] Tests Passed
- [x] Coverage Reviewed
- [x] Feature worked as expected
- [x] Added or updated flags to `--help` and `README.md`
- [x] Is the version upgrade path clear for this change? (breaking vs minor vs
  patch)
- [x] Follow up work tracked in a card if needed

### CODE REVIEW

**Comment by RobertReaves** [comment]
Should we are about their origin branch or upstream branch being up to date if they already exist? For example, they push to their origin and then make a change locally and attempt to run `tag-release --dev` it won't push into their origin, instead it will just make a PR from their origin into the upstream feature-branch without their local change.

File: package-lock.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "tag-release",
-  "version": "5.14.0",
+  "version": "5.14.2",
```
Comment (elijahmanor) [inline package-lock.json]: Seems weird that it changes the version to what it should have been. We need to figure this out... so this doesn't continue to happen

File: specs/help.spec.js [hunk 2]
```diff
@@ -49,6 +49,7 @@ describe("help", () => {
     $ tag-release --qa myorg                # create host repo qa branch w/scope
     $ tag-release --pr                      # create host repo pr into develop
     $ tag-release --pr myorg                # create host repo pr w/scope
+    $ tag-release --dev                     # create origin branch into upstream branch pr
```
Comment (elijahmanor) [inline specs/help.spec.js]: nice job remembering to add this :)

File: src/git.js [hunk 3]
```diff
@@ -433,6 +446,25 @@ const git = {
 	status(showOutput = true) {
 		const args = "status";
 		return git.runCommand({ args, showOutput });
+	},
+
+	createBranchUpstream(branch, hasBase = false) {
```
Comment (elijahmanor) [inline src/git.js]: Maybe create a common `createRemoteBranch` and have these both use it?

File: src/utils.js [hunk 4]
```diff
@@ -85,6 +85,22 @@ export default {
 			});
 		});
 	},
+	editBody(data) {
```
Comment (elijahmanor) [inline src/utils.js]: Possibly make a common method called `editFile` and have both `editLog` and `editBody` use it?

File: src/workflows/steps/index.js [hunk 5]
```diff
@@ -467,17 +470,24 @@ export function githubOrigin(state) {
 	return util
 		.exec(command)
 		.then(data => {
-			state.remotes.origin.url = data;
+			if (state.hasOwnProperty("remotes")) {
```
Comment (elijahmanor) [inline src/workflows/steps/index.js]: Maybe use `lodash.set`?

File: src/utils.js [hunk 6]
```diff
@@ -69,34 +69,16 @@ export default {
...
-				} else {
-					reject(`Unable to edit ${tempFilePath}`);
-				}
-			});
-		});
-	},
-	editBody(data) {
-		const tempFilePath = "./.pullrequest";
-
+	editFile(data, filePath) {
```
Comment (elijahmanor) [inline src/utils.js]: We could probably get rid of `filePath` completely and generate that location...

`_.uniqueId('.editFile_');`

In our clean-up we'd need to remove (or unlinkSync) `.editFile_*`

**Pull request merged**